modules/tasks/tasks/src/main/java/org/motechproject/tasks/util/SchedulerTaskTriggerUtil.java

```diff
@@ -0,0 +1,203 @@
+                schedulerService.safeScheduleJob(
+                        new CronSchedulableJob(motechEvent, cronExpression, startDate.toDate(), endDate.toDate(),
+                                ignorePastFiresAtStart));
+                break;
+
+            case DAY_OF_WEEK_JOB:
+                LOGGER.info("Scheduling day of week job for task: {}", task);
+                startDate = DATE_FORMAT.parseDateTime(trigger.getStartDate());
+                endDate = DATE_FORMAT.parseDateTime(trigger.getEndDate());
+                days = trigger.getDays();
+                time = trigger.getTime();
+                ignorePastFiresAtStart = trigger.isIgnoreFiresignorePastFiresAtStart();
+                schedulerService.scheduleDayOfWeekJob(
+                        new DayOfWeekSchedulableJob(motechEvent, startDate.toLocalDate(), endDate.toLocalDate(),
+                                days, time, ignorePastFiresAtStart));
+                break;
+
+            case REPEATING_JOB_WITH_PERIOD_INTERVAL:
+                LOGGER.info("Scheduling repeating job with period interval for task: {}", task);
+                startDate = DATE_FORMAT.parseDateTime(trigger.getStartDate());
+                endDate = DATE_FORMAT.parseDateTime(trigger.getEndDate());
+                repeatPeriod = trigger.getRepeatPeriod();
+                ignorePastFiresAtStart = trigger.isIgnoreFiresignorePastFiresAtStart();
+                schedulerService.safeScheduleRepeatingPeriodJob(
+                        new RepeatingPeriodSchedulableJob(motechEvent, startDate.toDate(), endDate.toDate(),
+                                repeatPeriod, ignorePastFiresAtStart));
+            default:
+                break;
+        }
+    }
+
+    public void setSchedulerTaskTriggerType(Task task) {
+        String[] name = task.getTrigger().getSubject().split("\\.");
+        switch (name[name.length-1]) {
+            case "runOnceJob":
+                ((SchedulerTaskTriggerInformation) task.getTrigger()).setType(SchedulerJobType.RUN_ONCE_JOB);
+                break;
+
+            case "repeatingJob":
+                ((SchedulerTaskTriggerInformation) task.getTrigger()).setType(SchedulerJobType.REPEATING_JOB);
+                break;
+
+            case "cronJob":
+                ((SchedulerTaskTriggerInformation) task.getTrigger()).setType(SchedulerJobType.CRON_JOB);
+                break;
+
+            case "dayOfWeekJob":
+                ((SchedulerTaskTriggerInformation) task.getTrigger()).setType(SchedulerJobType.DAY_OF_WEEK_JOB);
+                break;
+
+            case "repeatingJobWithPeriodInterval":
+                ((SchedulerTaskTriggerInformation) task.getTrigger()).setType(SchedulerJobType.REPEATING_JOB_WITH_PERIOD_INTERVAL);
+                break;
+
+            default:
+                break;
+        }
+    }
+
+    private MotechEvent prepareSchedulerEvent(String subject) {
```

Comment:
don't see much point in this method, given that it can get shortened to one line